pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -200,6 +201,18 @@ type SecretOrConfigMap struct {
 	ConfigMap *v1.ConfigMapKeySelector `json:"configMap,omitempty"`
 }
 
+// PodMonitoringStatusContainer represents a Kubernetes CRD that monitors pods
+// and contains a status sub-resource.
+type PodMonitoringStatusContainer interface {
+	client.Object
+
+	// Returns a unique identifier for this CRD, that may be used as a map key.
+	GetKey() string
```

Comment:
I think the interface is still useful because it lets us type our status update queue in the collection reconciler but maybe it can be reworked/renamed some other way.